### DISCUSSION TITLE
Exercise 18: consensus and Paxos unsafety

### DISCUSSION THREAD

**Discussion by dag7dev** [sapienzastudentsnetwork/distributed-systems#43] [Exercises]
**Exercise 18:** Assume that you remove the property that every round is associate to a unique proposer. After collecting a quorum of $n-f$ promises (where $n$ is the number of acceptors and $f$ is such that $n=2f+1$), the proposer chooses one of the values voted in max round in the promises (of course it is not unique, the proposer chooses just one in an arbitrary way). Show that Paxos is not safe any more.

**Comment by susannacifani** [comment]
What do you think about the solution on Drive?
It looks kinda right, but I'm not really sure about that

![image](https://github.com/sapienzastudentsnetwork/distributed-system-2023-2024/assets/73530772/6de1702d-9ae7-48d6-a398-aaf196f9f25b)

**Comment by cornflexxx** [comment]
Suppose that a proposer reach a quorum $n-f$ of promises with some value $v$ and some value $v'$. In the same round another proposer reach a similar quorum (with some $v$ and some $v'$).  Both sends accept in round $j$, but one for $v$ and the other for $v'$. Only $v$ reaches the majority of the acceptor for first, instead $v'$ reaches only some of them (and is voted only by the acceptors who don't have already voted for $v$). Now consider round $j+1$ in which a proposer gets a quorum of promises with values $v$ and $v'$. According to the new rules, the proposer can send $\langle \text{ACCEPT } v' \rangle$ in one of the subsequent rounds, which can reach a majority of acceptors and be chosen by the learners.
